sharedbuffers/mapped_struct.py

```diff
@@ -864,6 +886,72 @@ def __getitem__(self, index):
 
         return res
 
+    @staticmethod
+    def _adjust_min(val, nmax, step):
+        val = min(val, nmax)
+        if step < 0 and val == nmax:
+            val -= 1
+        return val
+
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            start, end, step = index.start, index.stop, index.step
+            xlen = len(self)
+
+            # Lots of adjustments going on here in order to correctly
+            # set the start and end indexes.
+
+            if step is None:
+                step = 1
+            elif step == 0:
+                raise ValueError("slice step cannot be zero")
+
+            if start is None:
+                start = self.elem_start if step > 0 else xlen - 1
+            else:
+                start += self.elem_start
+                if start < 0:
+                    start = xlen + start
+                    if start < 0:
+                        start = -1 if step < 0 else 0
+                elif start >= xlen:
+                    start = xlen if step > 0 else xlen - 1
+
+            if end is None:
+                end = xlen if step > 0 else -1
+            elif end < 0:
+                end = xlen + end
+                if end < 0:
+                    end = 0 if step > 0 else -1
+
+            end = self._adjust_min(end, xlen, step)
+            start = self._adjust_min(start, xlen, step)
+
+            if step < 0:
+                if end >= start:
+                    return []
+                r_len = (start - end - 1) / (-step) + 1
+                end = (r_len - 1) * step + start - 1
+            elif start >= end:
+                return ()
```

Comment:
Make up your mind. You either return an empty list or an empty tuple.

In fact, proxied_list should return an empty list and proxied_tuple should return an empty tuple.